microsoft/testsuites/mdatp/check-mdatp.sh

```diff
@@ -0,0 +1,113 @@
+#! /bin/sh
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+# check the following folders for printable files, print them
+# checks Azure Extension directories using globs, since they're not 
+# well-known.
+
+# Returns nonzero value on detection of any defender info
+# Return value is a bitmask showing the types of info detected.
+# Bits are ordered by seriousness of the issue:
+# Least significant bit is bad
+# Most significant bit is worst.
+EXIT_CODE=0
+# if any mdatp/defender installation dir is found
+EXIT_MDATP_INSTALLED=1
+# if any log dirs are found
+EXIT_MDATP_LOGS_FOUND=2
+# if an installation log is found
+EXIT_MDATP_INSTALL_LOGS_FOUND=4
+# if an onboarding blog is found
+EXIT_ONBOARD_INFO_FOUND=8
+MDATP_OPT_DIR='/etc/opt/microsoft/mdatp'
+MDATP_LOG_DIR='/var/log/microsoft/mdatp'
+ERROR_MSG_HEADER="----------------------------------------------------------------------"
+
+find_printable_files () {
+    found_dir="$1"
+    sudo find "$found_dir" -type f -exec file '{}' ';' -exec grep -Iq . '{}' ';' -exec head -c 1024 '{}' ';'
+}
+
+check_unexpected_file () {
+    surprise_file="$1"
+    echo "Found unexpected regular file: $surprise_file" >&2
+    # see what it is, print if it's printable
+    file "$surprise_file"; grep -Iq . "$surprise_file" && cat "$surprise_file"
+}
+# check for mdatp extension, folder is version labeled, so use a shell glob
+for mde_dir in /var/lib/waagent/Microsoft.Azure.AzureDefenderForServers.MDE.Linux* ; do
+    # for all the versioned folders we find...
+    if [ -e "$mde_dir" ]; then
+        EXIT_CODE=$EXIT_MDATP_INSTALLED
+        if [ -d "$mde_dir" ]; then
+            # find regular files, skip printing them if they are binary
+            find_printable_files "$mde_dir"
+        else
+            # not expecting to find a regular file in /var/log/azure, but...
+            check_unexpected_file "$mde_dir"
+        fi
+    fi
+done
+
+# check for azure extension install dir
```

Comment:
check for **Mde agent** install dir
No extension dir

Reply:
Thanks! Fixed the comment